src/CommandIsolator.cpp

```diff
@@ -124,6 +130,10 @@ process::Future<ContainerLimitation> CommandIsolatorProcess::watch(
 
   JSON::Object inputsJson;
   inputsJson.values["container_id"] = JSON::protobuf(containerId);
+  if (m_infos.contains(containerId)) {
+    inputsJson.values["container_config"] =
+        JSON::protobuf(m_infos[containerId]);
+  }
```

Comment:
shouldn't we fail if config is not there? It seems to me that it would be a bug to have watch called before prepare actually ran. At least we should log something explaining the input file for watch will be different than usual

Reply:
According with one of their own example, prepare must have been called before watch/usage/... So I fill fix my code
[source](https://github.com/mesos/modules/blob/master/isolator/test_isolator_module.cpp)

Reply:
If we fail when `prepare` is not called before `watch` or `usage`, this make all the unit tests fail because that's how they were working... Should I fix all the UT ?

Reply:
I think we should but if you find this goes beyond the scope of this PR, let's make it second one. What do you think?

Reply:
I think this belongs to the same PR as this patch adds a new 'dependency' between the callbacks. I'm on it :)